relayer/listener.go

```diff
@@ -61,7 +70,34 @@ func NewListener(
 	relayerHealth *atomic.Bool,
 	cfg *config.Config,
 ) (*Listener, error) {
-	sub := vms.NewSubscriber(logger, sourceBlockchain)
+	blockchainID, err := ids.FromString(sourceBlockchain.BlockchainID)
+	if err != nil {
+		logger.Error(
+			"Invalid blockchainID provided to subscriber",
+			zap.Error(err),
+		)
+		return nil, err
+	}
+	ethWSClient, err := ethclient.Dial(sourceBlockchain.WSEndpoint)
```

Comment:
nit: since ethWSClient doesn't seem to be anywhere else in the file, we can create the client in `NewSubscriber`

Reply:
We pass in an ethClient in the constructor so that we can mock it in the unit tests.